\documentclass[a4paper,11pt]{article}
\usepackage[utf8]{inputenc}
\usepackage{hyperref}
\usepackage{xcolor} % Required for specifying custom colours

\colorlet{mdtRed}{red!50!black}
\hypersetup{
	colorlinks,
	citecolor=mdtRed,
	filecolor=black,
	linkcolor=mdtRed,
	urlcolor=mdtRed
}
\usepackage{color}
\usepackage{graphicx}
\usepackage{lineno} % for line numbering
\usepackage{setspace} % for double-spaced text
\newcommand{\hl}[1]{\textcolor{red}{#1}}
\usepackage[left=2.5cm, right=2.5cm, top=3cm, bottom=3cm]{geometry} % for margins
\usepackage{tikz}
\usetikzlibrary{arrows.meta, calc, fit, tikzmark, bayesnet}
\usepackage{amsmath,amsfonts}
\usepackage{bbm}
\usepackage{authblk}
\usepackage{sidecap}
\sidecaptionvpos{figure}{t}
\usepackage[labelfont=bf,font={it}]{caption}

\setcounter{figure}{-1}


%%%%%%%%%%%%%%%%
% MATH SYMBOLS
%%%%%%%%%%%%%%%%
\DeclareMathOperator{\Poisson}{Poisson}
\DeclareMathOperator{\Multinom}{Multinom}
\DeclareMathOperator{\logit}{logit}
\DeclareMathOperator{\cov}{cov}
\DeclareMathOperator{\Ind}{I}
\def\P{\mathbb{P}}
\def\E{\mathbb{E}}

\def\x{\boldsymbol{x}}

\def\balpha{\boldsymbol{\alpha}}
\def\btau{\boldsymbol{\tau}}
\def\bmu{\boldsymbol{\mu}}
\def\bxi{\boldsymbol{\xi}}

\def\bLambda{\boldsymbol{\Lambda}}
\def\bP{\boldsymbol{P}}
\def\bd{\boldsymbol{d}}


\def\Ccal{{\cal C}}
\def\C{{\cal C}}
\def\E{{\cal E}}
\def\I{{\cal I}}
\def\N{{\cal N}}
\def\M{{\cal M}}
\def\R{{\cal R}}
\def\T{{\cal T}}
\def\X{{\cal X}}
\def\Y{{\cal Y}}
\def\Z{{\cal Z}}
\def\D{{\cal D}}

%%%%%%%%%%%%%%%%
% References
%%%%%%%%%%%%%%%%

\usepackage[style=numeric,natbib=true,doi=true,isbn=false,url=false,uniquename=false,uniquelist=false,firstinits=true,bibencoding=utf8]{biblatex}
\AtEveryBibitem{\clearfield{month}}
\AtEveryCitekey{\clearfield{month}}
\AtEveryBibitem{\clearfield{pages}}
\AtEveryCitekey{\clearfield{pages}}
\AtEveryBibitem{\clearfield{archivePrefix}}
\AtEveryCitekey{\clearfield{archivePrefix}}
\AtEveryBibitem{\clearfield{arxivId}}
\AtEveryCitekey{\clearfield{arxivId}}
\AtEveryBibitem{\clearfield{eprint}}
\AtEveryCitekey{\clearfield{eprint}}


\DeclareNameFormat{newformat}{%
	\nameparts{#1}% split the name data, will not be necessary in future versions
	\usebibmacro{name:newformat}%
}

\bibliography{library.bib}
\setlength\parindent{0pt}
%\linenumbers

%%%%%%%%%%%%%%%%
% Opening
%%%%%%%%%%%%%%%%

\title{Annual Report Year 1 \\
	Anticipating the Chemistry of Life (ACOL)}

\date{\today} % Remove date
\author[1,2]{Marco Visani}
\affil[1]{Department of Biology, University of Fribourg, 1700 Fribourg, Switzerland}
\affil[2]{Swiss Institute of Bioinformatics, 1700 Fribourg, Switzerland}

\begin{document}

\maketitle

\tableofcontents

\section{Introduction}
Public databases compiling genes (e.g., GenBank) and proteins (e.g., UniProt) have long been essential resources for the life sciences. Until recently, no equivalent resource existed for small molecules and their producing organisms. The LOTUS\cite{rutzLOTUSInitiativeOpen2022} project was designed to address this issue, and forms currently the most comprehensive collection of natural products (NP) occurrences. It however gathers compounds physically isolated from species. While this ensures high confidence, the process of extraction and isolation is too slow to realistically document the chemical diversity of life. To address this issue, we explore two directions: experimental and computational. On one side, the Earth Metabolome Initiative will establish an open, linked knowledge base documenting the metabolome of all living organisms. In parallel, we develop computational strategies to anticipate the chemical constitution of species based on prior knowledge.

My PhD consists in developing a probabilistic model that integrates LOTUS with large-scale mass spectrometry datasets to predict species metabolomes and estimate the likelihood of encountering NPs across the tree of life.
After training and validation, I expect the model to predict the likelihood of encountering a given molecule in a given species, even if this compound has never been observed in this species before. I hope that this will help prioritize experimental efforts toward the most promising candidates and that this PhD will contribute to a better understanding of the distribution of specialized metabolites across the tree of life.

\section{Markov Random Field}
\subsection{WP 1}
Main project of my PhD.

We assume that the pattern of presence and absence is described by similarities within each dimension. For instance, closely related species may share a similar set of NPs and NPs related in their synthesis may share a similar distribution across species. To model such similarities, we adopt a \href{https://en.wikipedia.org/wiki/Markov_random_field}{Markov Random Field} (MRF) approach.

The first months were dedicated to my C++ training. I started with some simple implementations for the model (bitwise operations, class creation, etc.) and I familiarized myself with the tools developed in Daniel Wegmann's lab.

After having a first version of the model, we performed some simulations to check if the model could infer from data correctly. The simulations allowed us to find some errors in the inference, and make some adjustments to the model. We changed the following equations.
\begin{equation}
	\bLambda_c =
	\begin{pmatrix}
		-\alpha_c  & \alpha_c    \\
		1-\alpha_c & \alpha_c -1 \\
	\end{pmatrix}
	\label{eq:rate matrix}
\end{equation}
The rate matrix as now a single parameter $\alpha$ that models the change rate. This changes the transition probabilities between parent node $p(n)$ and $n$ to :

\begin{equation}
	\bP(n) = \exp(\bLambda_c \nu_c b(n)).
\end{equation}

Calculating the root state probabilities then gives :
\begin{equation}
	\label{eq:stationary distribution}
	\bP_{\infty} = \left(1-\alpha_c, \alpha_c\right).
\end{equation}

Furthermore, we added an error rate parameter to the probability of LOTUS given the Markov Field:

\begin{equation}
	\P(L_{ms}|\x(m,s), R_{ms}) =
	\begin{cases}
		\epsilon \quad   & \mathrm{if\ } \x(m,s)=0, L_{ms} = 1, \\
		1-\epsilon \quad & \mathrm{if\ } \x(m,s)=0, L_{ms} = 0, \\
		R_{ms} \quad     & \mathrm{if\ } \x(m,s)=1, L_{ms} = 1, \\
		1- R_{ms} \quad  & \mathrm{if\ } \x(m,s)=1, L_{ms} = 0.
	\end{cases}
\end{equation}

Finally, the probability of discovery $R_{ms}$ was changed to the following:

\begin{equation}
	R_{ms} = (1 - e^{-\gamma_0 P_m })  (1 - e^{-\gamma_1 Q_s})
\end{equation}

According to the grant, the first working package has been achieved in the correct time frame. We have a first working implementation of the model.
\subsection{WP 2}
The second working package consists in modelling and integrating mass spectrometry data into the model. We originally started with



Implementation of mass spec data still in discussion, but we are planning to move from ViMMS \cite{wandyViMMSFrameworkDevelop2022} to just using outputs of annotation tools. We might use a combination of Sirius, Metfrag, CFM-ID and maybe weight these outputs.

\subsection{Fake extract}
We would like to model the \textit{loss} of molecules during the extraction process from Nature to the data analysis.

As stated in the grant, we want to model the classes of analytes lost at the stages of extraction, liquid chromatography and mass spectrometry fragmentation. To do so, we have a bought a library of 550 pure natural products that we are going to mix together to simulate a biological extract. This will allow us to estimate the loss of molecules between the Nature and the mass spectrometry analysis. Ideally we would like to integrate this loss into the model.

Additionally we want to pass each pure compound in the mass spec. So far plate 3 was performed by a bachelor student, but results are weird, so we'll pass everything again. Plate 1 is done.
\section{LOTUS Expanded}
\subsubsection{Context}
One of my side projects is to perform the in silico expansion the molecules present in LOTUS. This allows to create molecules that are not present in LOTUS but that could be potential existing natural products. We then want to fragment in silico these molecules. We hope that this would allow to increase the number of annotations we are able to make on real data.
We hope that we will find some of the molecules we have generated in silico in brand-new research publication such as in the Journal of Natural Products.
So far, starting from $100'000$ compounds in LOTUS we generated $3\cdot10^6$ in silico compounds. As a first proof that the molecules generated are not completely random, there are about $17'000$ of the compounds that are found in COCONUT which is another natural products databases that doesn't report the biological occurrence of the structure. \\
Using the in silico molecules for MS2 annotation using the molecules from LOTUS would also allow to a certain taxonomical information aboout this in silco molecule.



\subsubsection{Task}
It turns out me generate some molecules that are true natural products so we would like to see if we can create this synergy between: in silco generation of a compound, create its in silico MS2 spectrum, try to detect them in real data, if we do, then expanded that molecule again.
\subsubsection{Object}
In silico chemical expansion of molecules in LOTUS, producing $3\cdot10^6$ molecules. Website is being build. We want to add the species to the visualization. Still need to figure out how this can be broadly used.

\section{EMI-Monorepo}
\subsubsection{Context}
EMI-Monorepo is our rust coded monolith were we are building all the tools for the Earth Metabolome Initiative. Our post doc Luca is leading the software development of the project. We are building everything in Rust from the backend to the frontend, to the data analysis part.

\subsubsection{Need}
Given the difficulty of the development of the portal, when Luca feels like a project could be a
good training for me to learn Rust, he asks me to do it.

\subsubsection{Task}
\subsubsection{Object}
Helped Luca on the Monorepo for a couple of crates. Lost a bit the track of what he is doing.

\section{Perspectives}

I'd like to continue working with Luca in Rust as it is a skill that I want to master by the end of the PhD.

\printbibliography

\end{document}
